### PR TITLE
[COS-1829] Sync organization from email data

### PR DESCRIPTION
**Pull request by grokkos**
## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

## Summary by CodeRabbit

- **New Features**
    - Added functionality to extract names from email addresses.
    - Introduced a new structure for handling email data including content, recipients, sender, and subject.
    - Implemented email synchronization services for both single and multiple emails, with comprehensive error handling and timeout management.
    - New route added for email synchronization in the web service.

### CODE REVIEW

File: packages/server/customer-os-webhooks/service/sync_email_service.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	if email.SentBy != "" {
+		domain := utils.ExtractDomain(email.SentBy)
+		domainsSlice = append(domainsSlice, domain)
+		name = utils.ExtractName(email.SentBy)
+	}
+
+	organizationData := model.OrganizationData{
+		BaseData: email.BaseData,
```
Comment (alexopenline) [inline packages/server/customer-os-webhooks/service/sync_email_service.go]: we may need to re-build BaseData for organization, to not include external id and external system

File: packages/server/customer-os-webhooks/service/sync_email_service.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		//CustomerOsId:   "", TODO check if this is needed
```
Comment (alexopenline) [inline packages/server/customer-os-webhooks/service/sync_email_service.go]: not needed

File: packages/server/customer-os-webhooks/model/email.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	Bcc         []string  `json:"bcc,omitempty"`
+	Cc          []string  `json:"cc,omitempty"`
+	Channel     string    `json:"channel,omitempty"`
+	ChannelData string    `json:"channelData,omitempty"`
+	Identifier  string    `json:"identifier,omitempty"`
+	EventType   string    `json:"eventType,omitempty"`
+	Hide        bool      `json:"hide,omitempty"`
+	BelongsTo   BelongsTo `json:"belongsTo,omitempty"`
+	// in sent to or sent by at least 1 contact should be available in the system
+	ContactRequired bool                   `json:"contactRequired,omitempty"`
```
Comment (alexopenline) [inline packages/server/customer-os-webhooks/model/email.go]: not sure we need them here, could be needed to construct when building sync for interaction app

File: packages/server/customer-os-webhooks/service/sync_email_service.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		Name:           name,
+		Domains:        domainsSlice,
+		Email:          email.SentBy,
```
Comment (alexopenline) [inline packages/server/customer-os-webhooks/service/sync_email_service.go]: remove Email

File: packages/server/customer-os-webhooks/service/sync_email_service.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+		domain := utils.ExtractDomain(email.SentBy)
+		domainsSlice = append(domainsSlice, domain)
+		name = utils.ExtractName(email.SentBy)
+	}
+
+	organizationData := model.OrganizationData{
+		BaseData: email.BaseData,
+		//CustomerOsId:   "", TODO check if this is needed
+		Name:           name,
+		Domains:        domainsSlice,
```
Comment (alexopenline) [inline packages/server/customer-os-webhooks/service/sync_email_service.go:63]: To be safe, let's enforce single value for slice

**Review by alexopenline — CHANGES_REQUESTED** [review]

File: packages/server/customer-os-webhooks/service/sync_email_service.go (hunk 1 above)
Comment (grokkos) [reply packages/server/customer-os-webhooks/service/sync_email_service.go]: can we leave as is for now as these two fields have the `omitempty` flag, so can be empty i.e for organization. What do you think?

**Review by alexopenline — APPROVED** [review]

**Pull request merged**